tests/test_virtual_chassis.py

```diff
@@ -24,3 +27,184 @@ def test_connectivity(self, vct):
                 _, out = dv.runcmd(['sh', "-c", "ping -c 5 -W 0 -q %s" % ip])
                 print(out)
                 assert '5 received' in out
+
+    def test_voq_switch(self, vct):
+        """ Test VOQ switch objects configuration """
+        dvss = vct.dvss
+        for name in dvss.keys():
+            dvs = dvss[name]
+            #Get the config info
+            config_db = DVSDatabase(swsscommon.CONFIG_DB, dvs.redis_sock)
+            metatbl = config_db.get_entry("DEVICE_METADATA", "localhost")
+
+            cfg_switch_type = metatbl.get("switch_type")
+
+            #Test only for line cards
+            if cfg_switch_type == "voq":
+                print("VOQ Switch test for {}".format(name))
+                cfg_switch_id = metatbl.get("switch_id")
+                assert cfg_switch_id != "", "Got error in getting switch_id from CONFIG_DB DEVICE_METADATA"
+
+                cfg_max_cores = metatbl.get("max_cores")
+                assert cfg_max_cores != "", "Got error in getting max_cores from CONFIG_DB DEVICE_METADATA"
+                
+                cfgspkeys = config_db.get_keys("SYSTEM_PORT")
+                sp_count = len(cfgspkeys)
+
+                asic_db = DVSDatabase(swsscommon.ASIC_DB, dvs.redis_sock)
+                keys = asic_db.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_SWITCH")
+                switch_oid_key = keys[0]
+                
+                switch_entry = asic_db.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_SWITCH", switch_oid_key)
+
+                value = switch_entry.get("SAI_SWITCH_ATTR_TYPE")
+                assert value == "SAI_SWITCH_TYPE_VOQ", "Switch type is not VOQ"
+                
+                value = switch_entry.get("SAI_SWITCH_ATTR_SWITCH_ID")
+                assert value == cfg_switch_id, "VOQ switch id is invalid"
+                
+                value = switch_entry.get("SAI_SWITCH_ATTR_MAX_SYSTEM_CORES")
+                assert value == cfg_max_cores, "Max system cores is invalid"
+                
+                value = switch_entry.get("SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST")
+                assert value != "", "Empty system port config list"
+                #Convert the spcfg string to dictionary
+                spcfg = ast.literal_eval(value)
+                assert spcfg['count'] == sp_count, "Number of systems ports configured is invalid"
+    
+    def test_chassis_app_db_sync(self, vct):
+        """ Test chassis app db syncing """
+        dvss = vct.dvss
+        for name in dvss.keys():
+            if name.startswith("supervisor"):
+                dvs = dvss[name]
+                chassis_app_db = DVSDatabase(swsscommon.CHASSIS_APP_DB, dvs.redis_chassis_sock)
```

Comment:
I am not sure how best to handle this, but we could probably add a similar `get_chassis_app_db` handler to DVS if it is a supervisor module. In the interest of unblocking other chassis PRs, I think directly initializing the DB here is OK and we can refine the initialization/getter part later.

Reply:
O.k. Agreed. Thanks.